EventsByTag.md

```diff
@@ -0,0 +1,95 @@
+# Events by tag
+
+WIP: This documents the new implementation, the existing one uses Cassandra Materialized
+views which have been reported to unstable. See [here](https://github.com/akka/akka-persistence-cassandra/issues/247)
+
+Events by tag is a hard query to execute efficiently in Cassandra. This page documents
+how it works and what are the consistency guarantees.
+
+## Bucket sizes 
+
+Events by tag partitions are grouped into buckets to avoid a partition for a given tag
+getting to large. 
+
+ 1000s of events per day per tag -- DAY
+ 100,000s  of events per day per tag -- HOUR
+ 1,000,000s of events per day per tag -- MINUTE
+ 
+This setting can not be changed after data has been written.
+ 
+More billion per day per tag? You probably need a specialised schema rather than a general library like this.
```

Comment:
Sorry for commenting on a closed PR, but is it really a limit of `1,000,000` per day per tag? Given that simple tentative operation, like transaction processing, emits at least 2 events per transaction per aggregate, it gives maximum supported throughput of `1000000/2/24/60/60=5.78 txns/s` which is very very low, not taking into account `Hold/Capture` and other scenarios, which may emit 3 or 4 events.

Reply:
No it is an order of magnitude for the cassandra schema we have. It is when it gets to billions i'd suggest designing a custom schema e.g. that compresses batches of events into a single column.

The old implementation put all tags for each day in a single cassandra partition. If a cassandra partition gets too large it starts to perform badly (and has a hard limit of 2 billion cells, a cell being similar to a column). Given we have ~10 columns in the table we actually had a hard limit before of 200million but it would have performed terribly before getting any where near this.

This PR supports dropping the bucket size down to a minute rather than a day meaning we have 1440 partitions per day. I think this solution will work fine until you have in the order of 10s of millions per minute.

Reply:
Ah, my bad! I mixed up `billion` and `million` :) Awesome work, thank you a lot.

Reply:
I added this to the page as it wasn't really clear where between the million and billion it would still work: https://github.com/akka/akka-persistence-cassandra/pull/275/files